js/style/style.js

```diff
@@ -67,12 +62,12 @@ function Style(stylesheet, animationLoop, options) {
 
         if (stylesheet.sprite) {
             this.sprite = new ImageSprite(stylesheet.sprite);
-            this.sprite.on('load', this.fire.bind(this, 'change'));
```

Comment:
Yes. `SpriteImage` fires a `style.change` event which is propagated to `Style` then `Map`.  The same flow occurs without this `on` -> `fire` interaction within `Style.`